generatorpipeline/accumulators.py

```diff
@@ -311,6 +314,55 @@ def lifetime(self, x):
         self._cov.lifetime = x
 
 
+class CacheAccumulator(Accumulator):
+    '''
+    A cache that implements the Accumulator interface. The value it returns will be the
+    last `length` elements, which have been added.
+    No calculation is performed.
+
+    length=1 the number of previous elements to return.
+
+    acc.value[-1] is the last element
+    acc.value[-2] is the one before the last element
+    and so on.
+    '''
+
+    def __init__(self, length=1):
+        self._n = 0
+        self._cache = deque(maxlen=length)
+        self._timecache = deque(maxlen=length)
+
+    def _accumulate_obj(self, obj):
+        self._n += 1
+        self._cache.append(obj)
+        # no need to remove any element, because `maxlen` is defined.
+        self._timecache.append(time.time_ns())
+
+    def _accumulate_other(self, other):
+        self._n += other.n
+        cache = deque(maxlen=self._cache.maxlen)
+        ctimes = deque(maxlen=self._cache.maxlen)
+        it1 = zip(self._timecache, self._cache)
+        it2 = zip(other._timecache, other._cache)
+        m = heapq.merge(it1, it2, key=lambda x: x[0])
+        for t, el in m:
+            cache.append(el)
+            ctimes.append(t)
+        self._cache = cache
+        self._timecache = ctimes
+
+    @property
+    def value(self):
+        '''
+        value[-1] is the last element.
+        '''
+        return list(self._cache)
```

Comment:
I'm wondering if we should work with a copy of the list here, as changing values here my lead to changes in the original data. See the attached notebook for an example.
[CacheAcc_Copy.pdf](https://github.com/skuschel/generatorpipeline/files/10350623/CacheAcc_Copy.pdf)

Reply:
you are right, I have been wondering about this as well and then concluded to not make a copy for three reasons:
1) We do not know what objects are in there. Could be dictionaries or lists or combinations or even user defined data types. Therefor we cannot guarantee that a proper copy is made. 
2) Its MUCH slower to make a copy. Our use case is for plotting so in favor of speed we rather dont copy.
3) If need the user can still make its own copy. Python generally doesnt copy if not required: if `x` is an `numpy.ndarray` then `a=x` also doent make a copy in python. `a=x` would make a copy in matlab, but this is not matlab ;)